addons/mail/controllers/link_preview.py

```diff
@@ -23,9 +23,10 @@ def mail_link_preview(self, message_id):
 
     @http.route("/mail/link_preview/hide", methods=["POST"], type="jsonrpc", auth="public")
     @add_guest_to_context
-    def mail_link_preview_hide(self, link_preview_ids):
+    def mail_link_preview_hide(self, link_preview_message_ids):
```

Comment:
It's used by the "delete all" link preview feature. It works because all the link previews sent by the client are always attached to the same message. 

But conceptually, you are right, we could have issues if you try to hide link previews from different messages. This issue exists before this PR.

Anyway, I updated the code to check `is_current_user_or_guest_author` for all link previews.